dds/DCPS/RTPS/Spdp.cpp

```diff
@@ -1774,58 +1775,11 @@ void Spdp::remove_agent_info(const DCPS::RepoId&)
 }
 #endif
 
-void
-Spdp::remove_expired_participants()
-{
-  // Find and remove any expired discovered participant
-  ACE_GUARD (ACE_Thread_Mutex, g, lock_);
-  // Iterate through a copy of the repo Ids, rather than the map
-  //   as it gets unlocked in remove_discovered_participant()
-  DCPS::RepoIdSet participant_ids;
-  get_discovered_participant_ids(participant_ids);
-  for (DCPS::RepoIdSet::iterator participant_id = participant_ids.begin();
-       participant_id != participant_ids.end();
-       ++participant_id)
-  {
-    DiscoveredParticipantIter part = participants_.find(*participant_id);
-    if (part != participants_.end()) {
-      const MonotonicTimePoint expr(
-        MonotonicTimePoint::now() -
-        rtps_duration_to_time_duration(
-          part->second.pdata_.leaseDuration,
-          part->second.pdata_.participantProxy.protocolVersion,
-          part->second.pdata_.participantProxy.vendorId
-        )
-      );
-      if (part->second.last_seen_ < expr) {
-        if (DCPS::DCPS_debug_level > 1) {
-          DCPS::GuidConverter conv(part->first);
-          ACE_DEBUG((LM_WARNING,
-            ACE_TEXT("(%P|%t) Spdp::remove_expired_participants() - ")
-            ACE_TEXT("participant %C exceeded lease duration, removing\n"),
-            OPENDDS_STRING(conv).c_str()));
-        }
-#ifdef OPENDDS_SECURITY
-        ICE::Endpoint* sedp_endpoint = sedp_.get_ice_endpoint();
-        if (sedp_endpoint) {
-          stop_ice(sedp_endpoint, part->first, part->second.pdata_.participantProxy.availableBuiltinEndpoints);
-        }
-        ICE::Endpoint* spdp_endpoint = tport_->get_ice_endpoint();
-        if (spdp_endpoint) {
-          ICE::Agent::instance()->stop_ice(spdp_endpoint, guid_, part->first);
-        }
-        purge_handshake_deadlines(part);
-#endif
-        remove_discovered_participant(part);
-      }
-    }
-  }
-}
-
 void
 Spdp::remove_discovered_participant_i(DiscoveredParticipantIter iter)
 {
-  ACE_UNUSED_ARG(iter);
+
+  remove_lease_expiration_i(iter);
```

Comment:
Does this need to recalculate the scheduled wait? Or is it sufficient to just let it wake up, decide that nothing happens and then reschedule?